### ISSUE TITLE
ExtUtils-Manifest is being published with a self-referential dependency

### ISSUE THREAD

**Issue by wphillipmoore** [wphillipmoore/efs-deploy#1]
When we built both ExtUtils-Manifest 1.69 and 1.70, they both ended up with dependencies on 1.69.   This results in being unable to download 1.69, since this is the worst kind of recursive dependency.   Note that these are "build" dependencies, so this is probably the direct result of trying to stay on the leading edge with the various build tools in the ExtUtils::* namespace.

First, we have to investigate why efsdeploy allowed the self-referential dependency in the first place.  Then, we need to either manually correct the published metadata on ftp.openefs.org, or rebuild and reupload the release with a patched efsdeploy.

**Comment by wphillipmoore** [comment]
Here's the basic problem....    When I built perl5/ExtUtils/Manifest/1.69-build001, the resulting dependencies did NOT include ExtUtils-Manifest, since it this was the FIRST release of that distro.   Therefore, any of the downstream build or test dependencies that wanted ExtUtils::Manifest, simply fell back to using the one bundled with the perl5 core releases.

When I built perl5/ExtUtils-Manifest/1.69-build002, now the 1.69 releasealias was present, so when we expand the dependencies for build and test, we end up picking this up.    Same problem when building 1.70, which ends up depending on 1.69.

This is a major problem, really.    I will need to figure out how to make the decision to use the latest release (whatever we find in /efs/dist/perl5), or use what we have in core, depend on the parent dependents, recursively.   Alternately, make the code prefer the core over cpan, unless a specific release has been requested, and the core can't satisfy it.   The latter approach would mean we end up with far fewer dependencies on all the build tools from CPAN, since most people don't specify a version dependency (not 100% sure that bit of intuition is correct, though).

 I am now convinced that it might make sense to rebuild the entire perl5 metaproj after rewriting these rules, so that we get rid of some of the dependency overhead.   First, we have to figure out how to prevent this problem from occuring.
